packages/indexer-agent/src/agent.ts

```diff
@@ -1046,16 +1034,19 @@ export class Agent {
         !deploymentInList(eligibleAllocationDeployments, deployment),
     )
 
-    // TODO: Same as before: Should we return early in here case reconciliation is not needed? Also,
-    // this conditional seems the be doing the same work as the previous one, we should consolidate
-    // them.
     if (deploy.length + remove.length !== 0) {
-      this.logger.info('Deployment changes', {
+      logger.info('Deployment changes', {
+        indexingNetworkSubgraph,
+        syncing: activeDeployments.map(id => id.display),
+        target: targetDeployments.map(id => id.display),
+        withActiveOrRecentlyClosedAllocation: eligibleAllocationDeployments.map(
+          id => id.display,
+        ),
         deploy: deploy.map(id => id.display),
         remove: remove.map(id => id.display),
       })
     } else {
-      this.logger.debug('No deployment changes are necessary')
+      logger.debug('No deployment changes are necessary')
```

Comment:
@fordN, I've merged the logs, maintaining the same condition: it will only be logged if deployment changes occur.